pkg/artifactory/datasource/datasource_artifactory_group.go

```diff
@@ -0,0 +1,62 @@
+package datasource
+
+import (
+	"fmt"
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-artifactory/v6/pkg/artifactory/resource/security"
+	"github.com/jfrog/terraform-provider-shared/packer"
+	"github.com/jfrog/terraform-provider-shared/predicate"
+	"github.com/jfrog/terraform-provider-shared/util"
+	"golang.org/x/net/context"
+	"net/http"
+)
+
+func ArtifactoryGroupSchema() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: DataSourceGroupRead,
+		Schema:      getDataSourceGroupSchema(),
+		Description: "Provides the Artifactory Group data source. Contains information about the group configuration and optionally provides its associated user list.",
+	}
+}
+
+func getDataSourceGroupSchema() map[string]*schema.Schema {
+	var includeUsersSchema = map[string]*schema.Schema{
+		"include_users": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     false,
+			Description: "Setting includeUsers to true will return the group with its associated user list attached.",
+		},
+	}
+
+	baseGroupSchema := security.GroupSchema
+	delete(baseGroupSchema, "detach_all_users")
```

Comment:
This will delete `detach_all_users` from the `security.GroupSchema` map since you've only make a reference, not a deep copy.

In this case, call `util.MergeMaps` first which deep copies, then delete the attribute.